src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_twinemperors.cpp

```diff
@@ -48,531 +48,418 @@ enum Misc
+
+            events.ScheduleEvent(EVENT_VEKNILASH_UPPERCUT, urand(14000, 29000));
+            events.ScheduleEvent(EVENT_VEKNILASH_UNBALANCINGSTRIKE, urand(8000, 18000));
+            events.ScheduleEvent(EVENT_SCARABS, urand(7000, 14000));
+            events.ScheduleEvent(EVENT_TELEPORT, 30000);
+            events.ScheduleEvent(EVENT_ENRAGE, 3600000);
+            events.ScheduleEvent(EVENT_HEAL, 1000);
         }
 
         void UpdateAI(uint32 diff)
         {
-            //Return since we have no target
-            if (!UpdateVictim())
-                return;
-
-            if (!TryActivateAfterTTelep(diff))
+            if (UpdateVictim() == false)
                 return;
 
-            //UnbalancingStrike_Timer
-            if (UnbalancingStrike_Timer <= diff)
-            {
-                DoCastVictim(SPELL_UNBALANCING_STRIKE);
-                UnbalancingStrike_Timer = 8000+rand()%12000;
-            } else UnbalancingStrike_Timer -= diff;
-
-            if (UpperCut_Timer <= diff)
-            {
-                Unit* randomMelee = SelectTarget(SELECT_TARGET_RANDOM, 0, NOMINAL_MELEE_RANGE, true);
-                if (randomMelee)
-                    DoCast(randomMelee, SPELL_UPPERCUT);
-                UpperCut_Timer = 15000+rand()%15000;
-            } else UpperCut_Timer -= diff;
-
-            HandleBugs(diff);
-
-            //Heal brother when 60yrds close
-            TryHealBrother(diff);
+            events.Update(diff);
 
-            //Teleporting to brother
-            if (Teleport_Timer <= diff)
+            while (uint32 eventid = events.ExecuteEvent())
             {
-                TeleportToMyBrother();
-            } else Teleport_Timer -= diff;
-
-            CheckEnrage(diff);
+                switch (eventid)
+                {
+                case EVENT_HEAL:
+                    HealBrotherIfCan();
+                    events.RepeatEvent(1000);
+                    break;
+                case EVENT_VEKNILASH_UPPERCUT:
+                    if (Unit* pUnit = SelectTarget(SELECT_TARGET_RANDOM, 0, NOMINAL_MELEE_RANGE, true))
+                        DoCast(pUnit, SPELL_UPPERCUT);
+                    events.RepeatEvent(15000 + rand() % 15000);
+                    break;
+                case EVENT_VEKNILASH_UNBALANCINGSTRIKE:
+                    DoCastVictim(SPELL_UNBALANCING_STRIKE);
+                    events.RepeatEvent(8000 + rand() % 12000);
+                    break;
+                case EVENT_SCARABS:
+                    if (Creature* pCreature = RespawnNearbyBugsAndGetOne())
+                        MutateBug(pCreature);
+                    events.RepeatEvent(2000);
+                    break;
+                case EVENT_TELEPORT:
+                    TeleportToBrotherIfCan();
+                    events.ScheduleEvent(EVENT_TELEPORT_AFTER, 2000);
+                    events.RepeatEvent(30000);
+                case EVENT_TELEPORT_AFTER:
+                    if (AfterTeleport == true)
```

Comment:
probably you do not need to put `== true`, but you can convert the variable into boolean using this:
`if (!!AfterTeleport)`